cinema-model/src/main/java/touk/recru/app/usecase/UseCase.java

```diff
@@ -0,0 +1,17 @@
+package touk.recru.app.usecase;
+
+import org.springframework.core.annotation.AliasFor;
+import org.springframework.stereotype.Component;
+
+import java.lang.annotation.*;
+
+@Target({ElementType.TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@Component
+public @interface UseCase {
```

Comment:
A jakbyś wyjaśnił, czym jest `UseCase`? :>

Reply:
Tak samo jak w przypadku UseCase jako adontacja jest tylko aliasem na Component + na pattern.
Klasy oznaczone UseCase będą wykonywały pewną logikę związana ze swoją nazwą i posiadały jedną metodę lub pare zoverloadowanych pod tą samą nazwą. Odpowiadają za funkcjonalność którą z pewnych powodów nie uważałem za poprawną, aby umieścić ją w konkretnym serwisie. Podczas implementowania jakiejś funkcjonalności czasami zdarza mi się sytuacja gdy potrzebuję wykonać opreację na paru serwisach ze sobą powiązanych. Zaimplementowanie takiej metody w jednym z serwisów zmniejsza czytelność kodu/szukanie funkcjonalności w przyszłości + zwiększa odpowiedzialność serwisu.
Use case'y tworzę aby bardziej zgadzał się z SOLID.
- S jako zmniejszenie odpowiedzialności w serwisach
- D jako iż jeden serwis jest niezależny od drugiego
Chociaż nie zawsze stosuję się do tej logiki, to jednak dążę, żeby jak najwięcej logiki tak wyglądała.
Zawsze chętnie nauczę się lepszych praktych, lub posłucham rad jak zrobić to lepiej :))